source/grid/grid_generator.cc

```diff
@@ -1910,6 +1910,259 @@ namespace GridGenerator
+      const Point<2>   bl(-outer_radius - pad_left, -outer_radius - pad_bottom);
+      const Point<2>   tr(outer_radius + pad_right, outer_radius + pad_top);
+      GridGenerator::subdivided_hyper_rectangle(
+        bulk_tria, step_sizes, bl, tr, colorize);
+
+      // now remove cells reserved from the cylindrical hole
+      std::set<Triangulation<2>::active_cell_iterator> cells_to_remove;
+      for (const auto &cell : bulk_tria.active_cell_iterators())
+        {
+          if ((cell->center() - center).norm() < outer_radius)
+            cells_to_remove.insert(cell);
+        }
+
+      Triangulation<2> tria_without_cylinder;
+      GridGenerator::create_triangulation_with_removed_cells(
+        bulk_tria, cells_to_remove, tria_without_cylinder);
+
+      const double tolerance = std::min(min_line_length(tria_without_cylinder),
+                                        min_line_length(cylinder_tria)) /
+                               2.0;
+
+      // set material ID in bulk part to something other than 2
+      for (const auto &cell : tria_without_cylinder.active_cell_iterators())
+        cell->set_material_id(1);
+
+      GridGenerator::merge_triangulations(tria_without_cylinder,
+                                          cylinder_tria,
+                                          tria,
+                                          tolerance);
+
+      // now set manifold ids:
+      for (const auto &cell : tria.active_cell_iterators())
+        {
+          // set all non-boundary manifold ids on the cells that came from the
+          // grid around the cylinder to the new TFI manifold id.
+          if (cell->material_id() == 2)
+            {
+              cell->set_manifold_id(tfi_manifold_id);
+              for (unsigned int face_n = 0;
+                   face_n < GeometryInfo<2>::faces_per_cell;
+                   ++face_n)
+                {
+                  const auto &face = cell->face(face_n);
+                  if (face->at_boundary() &&
+                      (face->center() - center).norm() <= outer_radius)
+                    face->set_manifold_id(polar_manifold_id);
+                  else
+                    face->set_manifold_id(tfi_manifold_id);
+                }
+            }
+          else
+            {
+              // ensure that all other manifold ids (including the faces
+              // opposite the cylinder) are set to the flat id
+              cell->set_all_manifold_ids(numbers::flat_manifold_id);
+            }
+        }
+
+      if (colorize)
+        for (auto &cell : tria.active_cell_iterators())
+          for (unsigned int face_n = 0;
+               face_n < GeometryInfo<2>::faces_per_cell;
+               ++face_n)
+            {
+              const auto face = cell->face(face_n);
+              if (face->at_boundary())
+                {
+                  const Point<2> center = face->center();
+                  // left side
+                  if (std::abs(center[0] - bl[0]) < 1e-10)
```

Comment:
Can we please not hard-code the tolerance, but rather create a constexpr variable?